charts/argo-cd/templates/argocd-server/ingress-grpc.yaml

```
@@ -19,41 +15,66 @@ metadata:
   name: {{ template "argo-cd.server.fullname" . }}-grpc
   labels:
     {{- include "argo-cd.labels" (dict "context" . "component" .Values.server.name "name" .Values.server.name) | nindent 4 }}
-{{- if .Values.server.ingressGrpc.labels }}
-{{- toYaml .Values.server.ingressGrpc.labels | nindent 4 }}
-{{- end }}
+    {{- if .Values.server.ingressGrpc.labels }}
+    {{- toYaml .Values.server.ingressGrpc.labels | nindent 4 }}
+    {{- end }}
 spec:
+  {{- if eq (include "argo-cd.ingress.apiVersion" $) "networking.k8s.io/v1" }}
+  {{- with .Values.server.ingress.ingressClassName }}
+  ingressClassName: {{ . }}
```

```suggestion
  {{- if .Values.server.ingress.ingressClassName }}
  ingressClassName: {{ .Values.server.ingress.ingressClassName  }}
```
I had problem when rewriting ingresses that it does not regocnize the field yet. This way we can ensure it only changes if set. Otherwise you have a conflict on the resource bcs you would also try to set it via annotations

---

Hmm and this is solved by using an `if` statement? The `with` block is not rendered by default (when the value is `nil` or empty string).

---

@mkilchhofer  is correct. With vs If makes no change here.